### DISCUSSION TITLE
How to include a file with dynamically generated name?

### DISCUSSION THREAD

**Discussion by chentianran** [quarto-dev/quarto-cli#11681] [Q&A] [answered]
### Description

Is there a way to include file whose name is dynamically generated? For example, I naively assumed the following would work:

````qmd
---
title: "Year end report"
format: pdf
engine: jupyter
---

```{python}
filename = "2024-report.qmd"
```

{{< inclulde  `{python} filename` >}}

The end.
````

But it does not. the `include` command require static file name. So is there a way to make this work in Quarto?

---
Before you suggest something like this...

```python
#| echo: false
with open(filename, 'r') as file:
    content = file.read()
    print(content)
```

in a Python code block, I should note that this would not work as the content is rendered as static raw text.

**Comment by mcanouil** [comment]
Please edit the title of your message to a meaningful one.
Thanks you!

**Comment by cscheid** [comment, marked as answer]
Include shortcodes are resolved before engines execute the code, so no, that's not a syntax that will work in Quarto.

**Reply by mcanouil** [reply]
This kind of generated file/code would need to occur in a pre-render script, see https://quarto.org/docs/projects/scripts.html#pre-and-post-render


Just in case, but I believe `include` shortcodes are the first shortcodes to be resolved so using meta/var shortcodes should not work as well.
